user-operator/src/test/java/io/strimzi/operator/user/operator/SimpleAclOperatorIT.java

```diff
@@ -0,0 +1,233 @@
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hamcrest.collection.IsEmptyCollection;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Arrays.asList;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+
+@ExtendWith(VertxExtension.class)
+public class SimpleAclOperatorIT {
+
+    private static final Logger log = LogManager.getLogger(SimpleAclOperatorIT.class);
+    private static final int TEST_TIMEOUT = 60;
+
+    private static Vertx vertx;
+
+    private static KafkaCluster kafkaCluster;
+
+    private static SimpleAclOperator simpleAclOperator;
+
+    private static Properties kafkaClusterConfig() {
+        Properties config = new Properties();
+        config.setProperty("authorizer.class.name", "kafka.security.auth.SimpleAclAuthorizer");
+        config.setProperty("super.users", "User:ANONYMOUS");
+        return config;
+    }
+
+    @BeforeAll
+    public static void beforeAll() {
+        vertx = Vertx.vertx();
+
+        try {
+            kafkaCluster =
+                    new KafkaCluster()
+                            .usingDirectory(Testing.Files.createTestingDirectory("simple-acl-operator-integration-test"))
+                            .deleteDataPriorToStartup(true)
+                            .deleteDataUponShutdown(true)
+                            .addBrokers(1)
+                            .withKafkaConfiguration(kafkaClusterConfig())
+                            .startup();
+        } catch (IOException e) {
+            assertThat(false, is(true));
+        }
+
+        simpleAclOperator = new SimpleAclOperator(vertx,
+                new DefaultAdminClientProvider().createAdminClient(kafkaCluster.brokerList(), null, null, null));
+    }
+
+    @Test
+    public void testNoAclRules(VertxTestContext context) {
+        Set<SimpleAclRule> acls = simpleAclOperator.getAcls("no-acls-user");
+        context.verify(() -> {
+            assertThat(acls, IsEmptyCollection.empty());
+        });
+        context.completeNow();
+    }
+
+    @Test
+    public void testCreateAclRule(VertxTestContext context) throws InterruptedException {
+        SimpleAclRule rule = new SimpleAclRule(
+                AclRuleType.ALLOW,
+                new SimpleAclRuleResource("my-topic", SimpleAclRuleResourceType.TOPIC, AclResourcePatternType.LITERAL),
+                "*",
+                AclOperation.READ);
```

Comment:
I guess we ought to have broader coverage of the space than just this. Can we try creating a whole bunch of rules with different rule types, operations, resource types and pattern types? Maybe try a parameterized test?

Reply:
I was wondering how much is it important for what we are testing here. 
We are not testing that the rule is really enforced after its creation, I mean we are not starting a real client authenticating with the user and trying to read from that topic; we are testing that the rule is created and this operation should be independent from what kind of rule we are creating (on topic, group, ... read, write, ...).